### ISSUE TITLE
Sostituire <em> (enfasi) per identificare un testo descrittivo

### ISSUE THREAD

**Issue by Fupete** [italia/bootstrap-italia#764]
## Versione di Bootstrap Italia

2.x

## Comportamento atteso

Rispettare il più possibile la corretta semantica dei tag HTML.

## Comportamento attuale

La semantica HTML corretta del tag `<em></em>` indica una enfasi al testo che racchiude.

È corretto quindi usarlo come esempio nella documentazione invece per definire lo stile visivo dell'[esempio lista con testo descrittivo](https://italia.github.io/bootstrap-italia/docs/organizzare-i-contenuti/liste/#con-testo-aggiuntivo-azioni-multiple-e-metadata)? A occhio starebbe semanticamente meglio più sul **titolo** che sulla _descrizione_ di ogni item no? 

![esempio lista con testi descrittivi](https://user-images.githubusercontent.com/6736103/199964911-410a0392-4c26-4dc1-a8fd-de8c24ab98fe.png)

![markup esempio](https://user-images.githubusercontent.com/6736103/199965053-db0da300-1217-4a87-b144-a804dc2ee56c.png)

## Possibili soluzioni

Cambiare la descrizione da tag `<em></em>` verso uno `<span></span>` dedicato? 
Domandarsi se il titoletto sopra dovrebbe essere racchiuso nel tag `<em></em>` per dargli semanticamente enfasi, a prescindere dalla visualizzazione **bold** da dare via css.

cc @astagi @cfabry

**Comment by gpeirolo** [comment]
@Fupete eviterei di mettere un `<em>` su ogni titoletto. La soluzione dello `<span>` mi sembra la più corretta, magari con una classe se ci fossero già selettori in conflitto.
Lascerei comunque lo stile CSS applicato all'`<em>` nel codice per evitare breaking changes.
Attendo conferma per procedere al fix.

**Comment by Fupete** [comment]
Ok. Coinvolgerei @astagi e @cfabry nel thread. 
Unica cosa che mi preoccupa è che in effetti quel titoletto non avrebbe nessuna semantica che lo identitichi come titoletto/bold rispetto alla descrizione sotto che ora è `<em>` e che correttamente faremo diventare eg. `<span class="description">`. Pensandolo letto da uno screen reader, sarebbe di seguito senza neppure punteggiatura per come è scritto attualmente no? 
Concordo che sarebbe da evitare breaking-changes se ci riusciamo. O da rimandare la parte breaking al futuro in caso.

**Comment by AlessandroVecchi** [comment]
Ciao,
Scusate se mi intrometto ma secondo me è sbagliato usare un <em> e forzarlo ad essere visualizzato come testo normale.

**Comment by Fupete** [comment]
È in parte infatti l'oggetto della issue @AlessandroVecchi, attualmente è così. La descrizione, non bold, è formattata da un tag `em`. E il titoletto, che avrebbe il vero bisogno di enfasi semantica, magari col semplice **bold**, non lo è.

**Comment by Fupete** [comment]
A11y review failed. I close this.

**Comment by Fupete** [comment]
@cfabry dice che in realtà l'unico modo di avere markup semantico sarebbe che il testo "titolo" fosse un heading, ne `em` come ora sulla descrizione, ne `span` al suo posto come provato nella PR appena chiusa, sono semanticamente corretti. Bisognerebbe usare un heading `h` sul titolo, e in quel caso lo <span class="text" usarlo solo per il sottotitolo descrittivo. L'`h` corretto, ovviamente, a seconda della posizione in pagina. Mi sembra un intervento _breaking_ @gpeirolo @astagi cosa ne pensate? Idee? 🤔

**Comment by cfabry** [comment]
Si, quel componente visto così non può che essere interpretato con una Hx. Così come è adesso, semanticamente, la stringa è unica, sia con span che con em. Tuttavia visivamente sembrano due stringhe. La struttura di base invece dovrebbe essere <hx>[titolo]</hx><p>[paragrafo]</p>

**Comment by gpeirolo** [comment]
@Fupete mi sembra problematico più che altro perché dovremmo lasciare la possibilità di inserire un heading h2/h6 (dipende dal livello del contenuto), gestendo quindi tutti gli override del caso in CSS. Ma forse è più delicata la parte dell'editor che potrebbe trovarsi davanti all'imbarazzo della scelta. Inoltre varrebbe solo per quel caso specifico dato che non credo si voglia mettere come heading qualsiasi label di una lista.

**Comment by cfabry** [comment]
@gpeirolo sarebbe anche sbagliato gestire override dello stile in base a ciò che si vede. Ossia, anche il suggerimento stile Bootstrap del tipo `<h2 class="h5">` sarebbe sbagliato. A mio avviso bisogna lasciare nella demo un `<Hx>[titolo]</Hx><p>[paragrafo</p>` e come al solito suggerire di usare il livello di H appropriato. Mi rendo conto è completamente diverso da prima ma così sarebbe risolutivo.

**Comment by Fupete** [comment]
Lato versionamento diventa una `breaking-change` importante di questa variante di lista detta così. Diventerebbe la prima major 2.1 corretto @astagi ? Valutiamo bene. 

Lato semantica mi torna @cfabry quel che dici, e magari raccogliere `hx` e `p` magari dentro un `div` che li contenga. Valutarei se al posto di p possiamo lasciare lo `span class="text"` per essere meno breaking. E aggiungere l'`h` prima come titolo e il `div` a contenerli. Posso fare una PR più tardi per provarla. 

Però sul non "ridimensionare" l'`h` con le classi ad hoc non sono convinto. Per funzionare lato utente questa lista, utile per es. a fare elenchi di file o link esterni con le loro azioni a lato, non è secondo me sostenibile che abbia tanti titoli eg. h3 grandi con magari sotto il testino caption. Diventerebbe inusabile no? Semanticamente corretta ma decisamente poco usabile.

**Comment by astagi** [comment]
@Fupete sto prevedendo una major release, massimo entro la prossima settimana, se puoi aprire una PR relativa a questo la includiamo in questa major.

**Comment by Fupete** [comment]
@astagi @cfabry Ho aperto una PR dedicata https://github.com/italia/bootstrap-italia/pull/816, nei commenti lì qualche cosa ancora da fare. Prossimi giorni la guardiamo insieme se volete.

**Comment by Fupete** [comment]
@stale take it easy ;-)

**Comment by AlessandroVecchi** [comment]
> @Stale take it easy ;-)

Oggi mi sono arrivate le notifiche di almeno 10 issue chiuse per inattività.
